builtin/credential/approle/validation.go

```diff
@@ -368,9 +368,12 @@ func (b *backend) nonLockedSecretIDStorageEntry(s logical.Storage, roleNameHMAC,
 		if result.SecretIDNumUses == 0 ||
 			result.SecretIDNumUsesDeprecated < result.SecretIDNumUses {
 			result.SecretIDNumUses = result.SecretIDNumUsesDeprecated
+			persistNeeded = true
```

Comment:
I'd pop a TODO above this block just so if we're grepping for TODOs it'll show up as something to take care of.